faiss/impl/pq4_fast_scan.cpp

```diff
@@ -58,8 +58,14 @@ void pq4_pack_codes(
         return;
     }
     memset(blocks, 0, nb * nsq / 2);
+#if !defined(_MSC_VER) && \
```

Comment:
Would you mind moving this logic to `platform_macros.h` and `#define FAISS_BIG_ENDIAN` so that the test is simpler in the actual code?

Reply:
Sure. Will do this and commit again.